### ISSUE TITLE
Better error response for Pipeline Config API

### ISSUE THREAD

**Issue by vishaldevgire** [gocd/gocd#6017]
##### Issue Type
- Bug Report

##### Summary

Pipeline config ~~API v5~~ APIv6 and other APIs who accept environment variables in request body, fail with `http` status code 500 when you don't provide any of the two properties 'encrypted_value'  OR `value` for an environment variable

 <details><summary>Example: environment variable username</summary>
 <p>

```bash
curl 'https://localhost:8154/go/api/admin/pipelines' \
  -i \
  -k \
  -u 'admin:badger' \
  -H 'Accept: application/vnd.go.cd.v6+json' \
  -H 'Content-Type: application/json' \
  -X POST -d '{
  "group": "first",
  "pipeline": {
    "label_template": "${COUNT}",
    "lock_behavior": "none",
    "name": "up45",
    "template": null,
    "parameters": [],
    "environment_variables": [
      {
        "name": "username",
        "secure": true
      }
    ],
    "materials": [
      {
        "type": "git",
        "attributes": {
          "url": "test-repo",
          "destination": null,
          "filter": null,
          "invert_filter": false,
          "name": null,
          "auto_update": true,
          "branch": "master",
          "submodule_folder": null,
          "shallow_clone": false
        }
      }
    ],
    "stages": [
      {
        "name": "up42_stage",
        "fetch_materials": true,
        "clean_working_directory": false,
        "never_cleanup_artifacts": false,
        "approval": {
          "type": "success",
          "authorization": {
            "roles": [],
            "users": []
          }
        },
        "environment_variables": [],
        "jobs": [
          {
            "name": "up42_job",
            "run_instance_count": null,
            "timeout": null,
            "environment_variables": [],
            "resources": [],
            "tasks": [
              {
                "type": "exec",
                "attributes": {
                  "run_if": [],
                  "command": "ls",
                  "args": ""
                }
              }
            ],
            "tabs": [],
            "artifacts": [],
            "properties": null
          }
        ]
      }
    ],
    "tracking_tool": null,
    "timer": null
  }
}'
```

</p></details>

###### Basic environment details

* Go Version: `19.3`

##### Steps to Reproduce
1. Start GoCD server
2. Copy example request from description
3. Execute it on terminal

##### Expected Results
Should gracefully exit with status code 422, stating that one of the two (`value` or `encrypted_value`) must be present in request.

##### Actual Results
Fails with a status code 500

**Comment by GaneshSPatil** [comment]
Fixed as part of #5967 #6406 
Closing

**Comment by adityasood** [comment]
verified on 19.6.0-9425
